content/en/device-provisioning-service/client-library.md

```diff
@@ -50,49 +51,75 @@ void plgd_dps_shutdown(void);
 plgd_dps_context_t *plgd_dps_get_context(size_t device);
 
 /// @brief Set DPS manager callbacks.
-void plgd_dps_manager_init_callbacks(plgd_dps_context_t *ctx, plgd_dps_on_status_change_cb_t on_change_cb,
-                                     void *on_change_data, oc_cloud_cb_t on_cloud_change_cb,
-                                     void *on_cloud_change_data);
+void plgd_dps_set_manager_callbacks(plgd_dps_context_t *ctx, plgd_dps_on_status_change_cb_t on_change_cb,
+  void *on_change_data, oc_cloud_cb_t on_cloud_change_cb, void *on_cloud_change_data);
 
 /**
  * @brief Start DPS manager to provision device by given server.
  *
- * Setup context, handlers, dps server and start dps manager.
+ * Setup context, handlers, dps endpoint and start dps manager.
  *
- * @param ctx device context (cannot be NULL)
+ * @param ctx dps context (cannot be NULL)
  * @param endpoint endpoint of the provisioning server (cannot be NULL)
  * @return 0 on success
  * @return -1 on failure
  */
-int plgd_dps_manager_start(plgd_dps_context_t *ctx, const char *endpoint);
+int plgd_dps_manager_start(plgd_dps_context_t *ctx);
 
 /**
  * @brief Stop DPS manager.
  *
  * Deregister handlers, clear context, stop manager and close connection.
  *
- * @param ctx device context (cannot be NULL)
+ * @param ctx dps context (cannot be NULL)
  */
 void plgd_dps_manager_stop(plgd_dps_context_t *ctx);
 
 /**
- * @brief Clean-up and restart DPS provisioning on factory reset.
+ * @brief Restart DPS manager to provision device by given server.
+ *
+ * Setup context, handlers, dps endpoint and start dps manager.
+ *
+ * @param ctx dps context (cannot be NULL)
+ * @param endpoint endpoint of the provisioning server (cannot be NULL)
+ * @return 0 on success
+ * @return -1 on failure
+ */
+int plgd_dps_manager_restart(plgd_dps_context_t *ctx);
+
+/**
+ * @brief Clean-up of DPS provisioning and stopping of DPS manager on factory reset.
  *
- * @param ctx device context (cannot be NULL)
+ * @param ctx dps context (cannot be NULL)
+ * @return 0 on success
+ * @return -1 on failure
  */
 int plgd_dps_on_factory_reset(plgd_dps_context_t *ctx);
```

Comment:
What exactly is cleaned up? What is cleaned up on IoTivity-stack? Is the device factory reset triggered? Is it disowned? Or only disconnected from the cloud? ACLs?

Reply:
from the factory reset handler? of iotivity? 
If yes, isn't iotivity reset to factory defaults removing storage folder on it's own? Just curious.

Reply:
No, the factory reset doesn't remove the storage folder. It resets Iotivity internal stuff (creds, acls, ael, sdi, doxm) to defaults and saves these defaults to storage (if storage was configured).